### ISSUE TITLE
rule import/no-unresolved support `exports` field in package.json

### ISSUE THREAD

**Issue by aladdin-add** [import-js/eslint-plugin-import#2495]
```js
import '@dian/dn/serve'
```

the package `@dian/dn` has an exports like:
```json
  "exports": {
    ".": "./dist/mod.js",
    "./actuator": "./dist/actuator/app.js",
    "./monitor": "./dist/monitor/index.js",
    "./middlewares": "./dist/middlewares/index.js",
    "./serve": "./dist/serve.js"
  },
```

it's working as expected, but eslint is reporting an error:
```bash
1:8  error  Unable to resolve path to module '@dian/dn/serve'  import/no-unresolved
```

**Comment by ljharb** [comment]
This plugin uses `resolve`, which does not yet support `exports`.

It's a best practice, for backwards compatibility, to have your exports LHS map to a RHS that pre-exports node would be able to resolve - doing so would make things Just Work with this plugin in the meantime.

**Comment by ljharb** [comment]
Possible duplicate of #2132.

**Comment by Airkro** [comment]
Try another resolver:
https://gist.github.com/danielweck/cd63af8e9a8b3492abacc312af9f28fd

**Comment by KubaJastrz** [comment]
duplicate of #1810
